pkg/ebpf/dns_cache.go

```diff
@@ -69,6 +76,9 @@ func (c *reverseDNSCache) Add(translation *translation, now time.Time) bool {
 		if ok {
 			val.expiration = exp
 			val.merge(translation.dns)
+			if extra := val.shrinkToSize(c.maxDomainsPerIP); extra > 0 {
```

Comment:
Instead of merging first and then getting rid of extra ones, can we drop new domains during the `merge` step if we have already hit the limit?